packages/chat-component-bindings/src/hooks/TypeAssertions.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ErrorBar, MessageThread, ParticipantList, SendBox, TypingIndicator } from '@internal/react-components';
+import { ChatParticipantListSelector } from '../chatParticipantListSelector';
+import { ErrorBarSelector } from '../errorBarSelector';
+import { MessageThreadSelector } from '../messageThreadSelector';
+import { SendBoxSelector } from '../sendBoxSelector';
+import { TypingIndicatorSelector } from '../typingIndicatorSelector';
+import { getSelector } from './usePropsFor';
+import React from 'react';
+
+/**
+ * This function is a compile type check that {@link GetSelector} returns
+ * values of the correct type.
+ *
+ * @private
+ */
+export function assertGetSelectorTypes(): unknown {
+  const chatParticipantListSelector: ChatParticipantListSelector = getSelector(ParticipantList);
+  const errorBarSelector: ErrorBarSelector = getSelector(ErrorBar);
+  const messageThreadSelector: MessageThreadSelector = getSelector(MessageThread);
+  const sendBoxSelector: SendBoxSelector = getSelector(SendBox);
+  const typingIndicatorSelector: TypingIndicatorSelector = getSelector(TypingIndicator);
```

Comment:
Do we need to manually assign types here? Like
const typingIndicatorSelector: TypingIndicatorSelector = getSelector(TypingIndicator);

Can we do 
const typingIndicatorSelector= getSelector(TypingIndicator);

the return value should be inferenced correctly, then we do the final comparation in return value?

Reply:
The concern here is sometimes, if getSelector return 'any' type by some mistakes, it will be still inferenced correctly if we do the type assignment at the very beginning

Reply:
Type inference via return type and via local variable should be identical, no?

Note that I'm _not_ casting the return type, this is just checking the inferred type.

Reply:
If the type is 'any' it will pass the type check after you cast it to anything like
const foo: any = ...;
const bar: AType = foo;

This will still work

Reply:
So the issue then is even though we are doing the type assertions with the big union type above, getSelector can still return a 'any' type?

Reply:
talked IRL and determined that the existing type assertions are enough.